app/components/AiChatBase.tsx

```diff
@@ -0,0 +1,156 @@
+'use client';
+
+import Image from 'next/image';
+import { PropsWithChildren, useEffect, useState } from 'react';
+import * as Dialog from '@radix-ui/react-dialog';
+import { cn, MarkdownRenderer, ScrollArea } from '@/app/components';
+import aiStarsSvg from '@/public/ai-stars.svg';
+import wizardSvg from '@/public/pixel-wizard.svg';
+import { Icon, IconButton, Input, Tag } from '@swapr/ui';
+import { twMerge } from 'tailwind-merge';
+import { FA_EVENTS } from '@/analytics';
+import { trackEvent } from 'fathom-client';
+
+interface AiBaseProps extends PropsWithChildren {
+  messages: any[];
+  trackOnClickEvents: string[];
+  isChatOpen?: boolean;
+}
+
+const MARKETING_LINK =
+  'https://swpr.notion.site/Presagio-AI-Predictor-Chatbot-Beta-Launch-f4ccdfa867e949d3badf10705b7c90aa';
+
+export const AiChatBase = ({
+  children,
+  messages,
+  trackOnClickEvents,
+  isChatOpen,
+}: AiBaseProps) => {
+  const [isOpen, setOpen] = useState(isChatOpen);
+  const [scrollAreaElement, setScrollAreaElement] = useState<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (scrollAreaElement) scrollAreaElement.scrollTop = scrollAreaElement.scrollHeight;
+  }, [messages, isOpen, scrollAreaElement]);
+
```

Comment:
Maybe we could do something like this (with another color scheme) as a small headlight to make the chat icon easier to detect. On the main page, it kinda fades away with the footer

```suggestion
  useEffect(() => {
    const timerA = setTimeout(() => {
      setBorderAndBgColor('border-surface-surface-1 bg-outline-warning-high-em');
    }, 500);
    const timerB = setTimeout(() => {
      setBorderAndBgColor(DEFAULT_BORDER_AND_BG_COLOR);
    }, 1000);

    return () => {
      clearTimeout(timerA);
      clearTimeout(timerB);
    };
  }, []);
  
```

Reply:
this is not related to the logo chat trigger, this is to scroll down messages inside of chatbox